wordpress/version-manager.js

```diff
@@ -0,0 +1,681 @@
+		cfg[key] = spl[1];
+	}
+
+	// Assign defaults where possible
+	if ( null === cfg.WORKING_DIR ) {
+		cfg.WORKING_DIR = getDefaultWorkingDir();
+	}
+
+	if ( null === cfg.REPOSITORY_URL ) {
+		cfg.REPOSITORY_URL = `https://wpcomvip-bot:${cfg.GITHUB_OAUTH_TOKEN}@github.com/Automattic/vip-container-images.git`;
+	}
+}
+
+/**
+ * Support for MacOS or Linux file system.
+ */
+function getDefaultWorkingDir() {
+	switch( process.platform ) {
+		case 'darwin': {
+			return `/Users/${username}/.local/share/vip/vip-container-images/version-manager`;
+		}
+		case 'linux': {
+			return `/home/${username}/.local/share/vip/vip-container-images/version-manager`;
+		}
+		default: {
+			console.log( 'Unsupported Operating System. Currently this script only supports MacOS and Linux' );
+			process.exit( 1 );
+		}
+	}
+}
+
+/**
+ * Attempts to organize the list of tags in an intelligent way.
+ * Show all editions of the current and previous major version
+ * Show only the the most recent point releases of previous major versions
+ */
+function collateTagList( tags, size ) {
+	const versionList = [];
+	let sizeOffset;
+
+	// sort tags
+	tags = tags.reverse();
+
+	// index tags
+	const { majorVersions, versions, releases } = indexTags( tags );
+
+	// build new tag list from indexes
+	OUT:
+	for ( const i in majorVersions ) {
+		sizeOffset = 0;
+		for ( const [ j, v ] of versions[ majorVersions[ i ] ].entries() ) {
+			// If it is the most recent 2 versions, append all of the releases
+			if ( i == 0 && j <= 0 ) {
+				sizeOffset += releases[ v ].length;
+				versionList.push( ...releases[ v ] );
+			} else {
+				// Append only the newest release for previous versions
+				// Signify it only with a x.x version
+				// Later it will be stapled to the latest release for that version
+				versionList.push( v );
+			}
+
+			if ( versionList.length >= ( size - sizeOffset) ) {
+				break OUT;
+			}
+		}
+	}
+
+	return { versionList, majorVersions, versions, releases };
+}
+
+/**
+ * Get list of queued adds
+ */
+function getAddsQueue( imageList, versionList, releaseIndex ) {
+	const adds = [];
+	let ref;
+
+	for ( version of versionList ) {
+		if ( imageList.indexOf( version ) === -1 ) {
+			ref = ( releaseIndex.hasOwnProperty( version ) ) ? releaseIndex[version][0] : version;
+			adds.push( { tag: version, ref: ref } );
+		}
+	}
+
+	return adds;
+}
+
+/**
+ * Get list of queued removes
+ */
+function getRemovesQueue( imageList, versionList, lockedList ) {
+	const removes = [];
+	for ( image of imageList ) {
+		if ( lockedList.indexOf( image ) === -1 ) {
+			if ( versionList.indexOf( image ) === -1 ) {
+				removes.push( { tag: image } );
+			}
+		}
+	}
+
+	return removes;
+}
+
+/**
+ * Get list of queued updates
+ */
+async function getUpdatesQueue( releases ) {
+	return new Promise( resolve => {
+		const updates = [];
+		let mostRecentRelease
+		// regex refType will match with N.N(.N)?
+		const refType = new RegExp( /[0-9]+\.[0-9]+(?:\.)?(?:[0-9]+)?/ );
```

Comment:
This regex can probably be simplified to something like (not tested beyond regex101.com): 
```suggestion
		const refType = new RegExp( /\d+\.\d+(\.\d+)?/ );
```

Reply:
That's good. Works great, thank you. I'll make this change.
![Screen Shot 2022-02-04 at 11 27 15 AM](https://user-images.githubusercontent.com/46167019/152583137-ab6ff8e3-91cd-402f-9b27-129e4a685541.png)

Reply:
I have added this suggestion to the most recent commit.